extensions/resteasy-classic/resteasy/deployment/src/test/java/io/quarkus/resteasy/test/ProviderConfigInjectionTest.java

```diff
@@ -47,11 +48,11 @@ public String getFoo() {
     public static class FooProvider implements ContextResolver<String> {
 
         @ConfigProperty(name = "foo")
-        String foo;
+        Instance<String> foo;
```

Comment:
Is this necessary? I know we tell people to do it, but does it not work otherwise with this PR?

Reply:
In my initial proposal it didn't work, because I was only adding the `BuildTimeRunTimeDefaultValuesConfigSource` plus the sources registered with the build step at static init (basically, there was not system, env or `application.properties` present).

In the end it become too much problematic to remove those sources. For instance, reactive messaging which is also static initialized, requires some configs in the `smallrye` namespace, but because we only record the `quarkus` namespace they are not available in the `BuildTimeRunTimeDefaultValuesConfigSource` and we need to search them in the other sources.

Additional to this, probably on a separate PR, we need to analyze which configuration properties are required for static init and either move them to the `quarkus` namespace or allow other namespaces to be recorded. Once we have that in place, then we can remove the system, env and `application.properties` from static int.

Reply:
I'm just concerned that this might break plenty of users

Reply:
Yes, that is a possibility if the configuration is in a user custom source and they use a direct injection in a Provider. I don't see a way to work around this.

Reply:
OK, let's just mark this a breaking change and make sure to add a note in the migration guide